clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5171,7 +5171,9 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
                                 /*Joined=*/true);
     else if (IsWindowsMSVC)
       ImplyVCPPCXXVer = true;
-    else if (IsSYCL)
+
+    if (IsSYCL && types::isCXX(InputType) &&
+        !Args.hasArg(options::OPT__SLASH_std))
```

Comment:
So is the difference in the behavior due to the MSVC driver not overriding the value when a new argument is encountered on the command line?

Reply:
Right .  We were setting up a default value (`-std=c++17`) for all platforms.  Due to the way the Windows MSVC path is taken (defaults set further down the line to better interact with `OPT__SLASH_std_EQ` we lost the ability to interject the user setting of `-std` and the default value was always set.